integration-tests/oidc-wiremock/src/test/java/io/quarkus/it/keycloak/SlackWiremockTestResource.java

```diff
@@ -0,0 +1,150 @@
+package io.quarkus.it.keycloak;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.containing;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlMatching;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+
+import java.io.Closeable;
+
+import org.jboss.logging.Logger;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.matching.AnythingPattern;
+
+public class SlackWiremockTestResource implements Closeable {
```

Comment:
This comment does not apply specifically to this file only... I wonder, if we should instead consider `oidc-wiremock-providers` and have a dedicated per-provider, Renarde-like way of testing it ? `oidc-wiremock` is a bit more generic, I use `github` here and there, but we already have like 20 tenants here, with OidcWiremock starting another server...

I'm also not sure about the necessity of creating such provider specific mocks on ports like `8188`.
May be we should instead update OIDC wiremock in `tests-support/oidc-server` to publish one more property like `oidc.url` instead of the current `keycloak.url`.

Similarly, I'm not sure about inlining all this key info, id tokens, OidcWiremock ships the key pair and it can generate ID tokens with required properties on the fly

Reply:
If you agree, we can look at it later in another PR

Reply:
A lot is going on in the OIDC wiremock tests, which is why you have to try adapt the Slack tenant to some other tests's requrements

Reply:
> This comment does not apply specifically to this file only... I wonder, if we should instead consider `oidc-wiremock-providers` and have a dedicated per-provider, Renarde-like way of testing it ? `oidc-wiremock` is a bit more generic, I use `github` here and there, but we already have like 20 tenants here, with OidcWiremock starting another server...
> 
> I'm also not sure about the necessity of creating such provider specific mocks on ports like `8188`. May be we should instead update OIDC wiremock in `tests-support/oidc-server` to publish one more property like `oidc.url` in addition to the current `keycloak.url`.
 
> Similarly, I'm not sure about inlining all this key info, id tokens, OidcWiremock ships the key pair and it can generate ID tokens with required properties on the fly

I intercepted responses and requests and tried to use them. This Slack is imperfect. I cannot decide whether using some generic mock won't mean we can miss something. I though we should keep that mock as close Slack as possible.

> If you agree, we can look at it later in another PR

I think I need to trust you here. I am honestly not able to tell if dropping this mock in favor of OIDC wiremock won't mean we will not test Slack provider properly.

Reply:
@sberyozkin   I had a look again (admittedly main reason was bullyish security tools telling me there is a leak [NO there isn't]) and I dropped hardcoded ID token. Better? If you are certain there is no value to added Slack mock resource, I can drop it altogether.

Reply:
Sorry @michalvavrik, I started the review and got distracted,  it is better, but I feel it would be much cleaner to have it in the dedicated `integration-tests/oidc-wiremock-providers`, where you just set `quarkus.oidc.provider=slack`, avoid hardcoding the port, and setting all of

```
quarkus.oidc.slack.tenant-paths=/slack
quarkus.oidc.slack.token.lifespan-grace=2147483647
quarkus.oidc.slack.authentication.force-redirect-https-scheme=false
quarkus.oidc.slack.authentication.verify-access-token=false
quarkus.oidc.slack.authentication.remove-redirect-parameters=false
quarkus.oidc.slack.authentication.scopes=oidc,profile,email
```

and will only have

```
quarkus.oidc.slack.provider=slack
quarkus.oidc.slack.auth-server-url=${oidc.url}
quarkus.oidc.slack.client-id=...
quarkus.oidc.slack.credentials.secret=secret
```

The other thing is that instead of updating the discovered metadata with the local addresses, one can add something like  `quarkus.oidc.slack.authorize-path=/authorize` which will resolve aganist OIDC wiremock provided `${oidc.url}`.

We have 10+ plus providers supported and if if we keep pushing that all into `oidc-wiremock`, it can become unmanageable...

As far as the utility of Slack specific and other provider wiremocks is concerned, it is probably rather limited, but it should be worth it with trying to emulate the precise Slack OIDC response structure from the discovery and token endpoints which is more useful for emulating OAuth2 servers, but ideally we'd reuse some stubs for managing the code flow...
For example, right now, the test you added does not really follow the code flow, as other tests do, this test misses a step where a user is challenged with the authentication screen.

IMHO we can make it cleaner and neater in a dedicated integration tests module...

Reply:
I agree that having `integration-tests/oidc-wiremock-providers` would be cleaner and this mock is not great. I tried to extract what should be tested. I don't know, so my conclusion is that I'll drop it and use OIDC Wiremock, but TBH, I don't understand if I am actually going to be testing "Slack" if I am using this generic stuff. I have to trust you there.


> For example, right now, the test you added does not really follow the code flow, as other tests do, this test misses a step where a user is challenged with the authentication screen.

That was intentional. I am testing Quarkus OIDC, not a code flow. I though this step does not interact with Quarkus OIDC. Maybe I put breakpoint at the wrong place, don't know.

Reply:
I will wait till you suggest what should be done (not exact configuration, just use OIDC Wiremock or whatever). I don't mind rewriting this completely. My motivation was to have `.well.known/openind-configuration` and `access-token` as it was issued to me (as well as `response_type`, `client_id`, `scope` query params). Maybe it is not necessary because every provider sends it, but then, we already know it works without testing? I'll let you decide.

Reply:
@michalvavrik Sure, let's keep it but how about us using this PR as an opportunity to create a dedicate integration test module for testing all of the well known providers we support, so that we can link users to it ?
Can you move the integration test into the new module ? I'd only like to ask to avoid starting another Wiremock server, as we will need to do it for every provider, have something like:

```
quarkus.oidc.slack.provider=slack
quarkus.oidc.slack.auth-server-url=${oidc.url}/slack
quarkus.oidc.slack.authorize-path=/authorize
quarkus.oidc.slack.token-path=/token
quarkus.oidc.slack.client-id=...
quarkus.oidc.slack.credentials.secret=secret
quarkus.oidc.slack.authentication.force-redirect-https-scheme=false
```

with `OidcWiremockServer` also publishing the wiremock server address as `oidc.url`, and you get the discovery and token stubs added with via the injected `@OidcWireMock` as is done for many of the code flow tests. And you can keep the discovery document exactly as is returned by Slack

Reply:
fine with me, thanks for details proposal

Reply:
I've done it according to what I understood should be done. If I misunderstood you, I'll deal with it tomorrow.